pkg/uploader/logger.go

```diff
@@ -0,0 +1,104 @@
+package uploader
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"unicode"
+)
+
+type Dclssue struct {
+	Line        int    `json:"line"`
+	Column      int    `json:"column"`
+	Message     string `json:"message"`
+	SyntaxError bool   `json:"syntaxError"`
+	Severity    string `json:"severity"`
+	Code        int    `json:"code"`
+}
+type CompileError struct {
+	Status       string   `json:"status"`
+	Message      string   `json:"message"`
+	Errors       []string `json:"errors"`
+	CompileError struct {
+		DclIssues map[string][]Dclssue `json:"dclIssues"`
+	} `json:"compile_error"`
+}
+
+const severityError = "ERROR"
+const severityWarning = "WARNING"
+
+func (up *uploader) logResponse(res *http.Response) {
+
+	if res.StatusCode == http.StatusOK || res.StatusCode == http.StatusCreated || res.StatusCode == http.StatusNotModified {
+		up.log.Info("Base DCLs uploaded and compiled successfully")
+		return
+	} else if res.StatusCode == http.StatusBadRequest {
+		var ce CompileError
+		err := json.NewDecoder(res.Body).Decode(&ce)
+		if err == nil {
+			err = up.printCompileError(ce)
+			if err == nil {
+				return
+			}
+		}
+	}
+	up.log.Error("unexpected response on DCL upload: status(%s) body(%s)", res.Status, res.Body)
+
+}
+func (up *uploader) printCompileError(compileError CompileError) error {
+	for file, issues := range compileError.CompileError.DclIssues {
+		for _, issue := range issues {
+			reader, err := os.Open(up.root + file)
```

Comment:
I think it would be better to read the whole file into a buffer once instead of for every issue (policy files are expected to be reasonably small).
This only as a remark, I'd be happy to leave it like this, since performance doesn't matter much here, IMO.

Reply:
IMO on compile error handling in staging phase is code readability more important than unnecessary file operations. Realistically there is not many compile errors. And even if... subseconds

Reply:
Makes sense

Reply:
You are right.

Reply:
You are right.